### DISCUSSION TITLE
SH-ESP32 unable to get onboard can tranceiver to work

### DISCUSSION THREAD

**Discussion by Unoxa** [hatlabs/discussions#100] [Q&A] [answered]
I have tried to use the SH-ESP32 as a gateway to convert NMEA2000 messages to NMEA0183 and forward them in a serial connection to an old plotter. I have used this project by Andreas Koritnik https://github.com/AK-Homberger/NMEA2000WifiGateway-with-ESP32 as a template and it works using the "Waveshare SN65HVD230 Can Board" as tranceiver on the SH-ESP32 with simulated NMEA2000 messages transmitted by ActiSenseListenerSender also using the Waveshare Can Board.
With this setup it works perfect including Wifi but I cannot get the onboard TI ISO1050 canbus tranceiver to work. 
Being new to this I would appreciate any tips, advice on how to proceed

**Comment by mairas** [comment]
Did you check that the CAN TX and RX GPIO pins are correct?

**Reply by Unoxa** [reply]
For Waveshare changed GPIO pins to 22 and 23, to use the onboard CAN GPIO 32 TX, 34 RX which did not work. Have tried both with and without 12V on the NMEA2000 connector. So I really don't know how to proceed, of course I can keep on using the Waveshare board but the TI ISO1050 would be better.
Any other suggestions?

**Reply by Unoxa** [reply]
An added question, is there any way to physically change GPIO pins for the ISO1050 transceiver on SH-ESP32? The reason for asking is that I noticed that GPIO 34 did not work with this sketch on another ESP32 WROOM 32D board.

**Reply by mairas** [reply]
You need to provide 12V to the NMEA 2000 header, otherwise the transceiver won't have power. And yes, you can remap the GPIOs by removing the 0R resistors and rewiring the pads according to your needs.

However, it would be much, MUCH easier to just fix the GPIO configuration in your software. Have a look e.g. at the [SH-ESP32-onewire-temperature](https://github.com/hatlabs/SH-ESP32-onewire-temperature/blob/main/src/main.cpp) code how it is done.

**Reply by Unoxa** [reply]
Thanx for quick reply. Checked the hardware by loading your SH-ESP32 NMEA 2000 gateway and getting it all to work. Will try to modify the software as it is a much better solution. However it will take time being new to this but I have time with with the boating season months away here in Stockholm! Might return with more questions.

**Reply by Unoxa** [reply, marked as answer]
Solved! My own mistake, GPIO 34 connected to another (unused) function in the code. Works now, only needs cleaning up. Thanks for your input.
